kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/FileSystemViewKeyInputFormat.java

```diff
@@ -76,24 +77,27 @@ public FileSystemViewKeyInputFormat(FileSystemDataset<E> dataset,
     Format format = dataset.getDescriptor().getFormat();
 
     boolean isSpecific = SpecificRecord.class.isAssignableFrom(dataset.getType());
+    String readerSchema = conf.get("kite.readerSchema");
+    if (isSpecific && readerSchema == null) {
```

Comment:
I think that this should require that the schema is null if `isSpecific` is true. You should be able to project columns by using a particular class (specific, reflect, thrift, etc.) or by setting a schema, but not both. We should not return partial objects: when I get an object each field should be what was stored in the dataset. This is especially important because Avro uses unsafe APIs to construct reflected objects so there are fewer guarantees.